sklearn/neighbors/graph.py

```diff
@@ -77,7 +79,13 @@ def kneighbors_graph(X, n_neighbors, mode='connectivity', metric='minkowski',
             ).fit(X)
     else:
         _check_params(X, metric, p, metric_params)
-    return X.kneighbors_graph(X._fit_X, n_neighbors, mode=mode)
+    if mode == "connectivity":
+        warnings.warn("The behavior of marking each sample itself as the "
```

Comment:
Let's say someone now writes new code, sees the method, and wants to use it.
Then they get a warning, telling them
 "this method is broken" and they will have to use a different method. Then why keep this method at all if they can't use it now?